src/field_theory/finite/trace.lean

```diff
@@ -18,10 +18,14 @@ finite field, trace
 
 namespace finite_field
 
+
 /-- The trace map from a finite field to its prime field is nongedenerate. -/
 lemma trace_to_zmod_nondegenerate (F : Type*) [field F] [finite F] {a : F}
- (ha : a ≠ 0) : ∃ b : F, algebra.trace (zmod (ring_char F)) F (a * b) ≠ 0 :=
+ (ha : a ≠ 0) :
+ by letI := zmod.algebra F (ring_char F);
```

Comment:
In Lean3, usually `haveI` is better as it doesn't produce a `let` in the lemma statement. I think letI and haveI are equivalent here in Lean 4.

Reply:
Doesn't `haveI` forget the definition, like `have` vs `let`?

Reply:
`haveI` works (I don't know if it forgets the definition or not, but maybe it doesn't matter, at the end the algebra structure is unique).

Reply:
HaveI only forgets the definition within the current tactic block, which is fine here